File: Hashing-Prob12.java
import java.util.*;
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> hash=new HashMap<>();
        for(String str: strs){
            char[] charArray=str.toCharArray();
            Arrays.sort(charArray);
            String sorted=String.valueOf(charArray);
            if(!hash.containsKey(sorted)){
                hash.put(sorted,new ArrayList<>());
            }
            hash.get(sorted).add(str);
        }
        return new ArrayList(hash.values());
    }
}

File: Hashing-Prob12.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict1 = dict()
        for string in strs:
            sortedString = "".join(sorted(string))
            if sortedString not in dict1.keys():
                dict1[sortedString] = list()
            dict1[sortedString].append(string)
        return dict1.values()
